archiver.go

```diff
@@ -1,33 +1,172 @@
 package debos
 
 import (
-	"log"
+	"fmt"
 	"os"
+	"path/filepath"
+	"strings"
 )
 
-func TarOptions(compression string) string {
+type ArchiveType int
+
+// Supported types
+const (
+	_ ArchiveType = iota // Guess archive type from file extension
+	Tar
+	Zip
+	Deb
+)
+
+type Archive struct {
+	File        string // Path to archive file
+	Type        ArchiveType
+	compression string
+}
+
+type Archiver interface {
+	CommandUnpack(destination string, options ...string) []string
+	GetType() ArchiveType
```

Comment:
Fwiw getters prefixed with Get isn't iodiomatic go. Also shouldn't the interface be Unpacker rahter then Archiver ? it doesn't archive anything :)

Reply:
Yet. But if we extend the interface with Pack() -- it would be.
Let me know if Unpacker is still preferable.

Reply:
I think it would be more idiomatic go if you then had a Pack interface next to the Unpack one; Like there are diferent interfaces for Reader and Writer